### PR TITLE
azure provider: replace UI instructions in provider setup with admin grant command

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

Our docs has UI instructions for this process but apparently there is a command in Azure CLI to do this.

### How has this code been tested?

 I've tested it in https://github.com/crossplane/stack-minimal-azure/pull/7 via Wordpress Stack.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation] and [examples].
- [x] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplane/crossplane/tree/master/design/one-pager-definition-of-done.md

### CODE REVIEW

File: docs/cloud-providers/azure/azure-provider.md [hunk 1]
```diff
@@ -41,26 +41,17 @@ You might see an error similar to the following, but that is OK, the permissions
...
 ```
 
-After these steps are completed, you should have the following file on your local filesystem:
-
-* `crossplane-azure-provider-key.json`
-
-## Grant Consent to Application Permissions
+Finally, you need to grant admin permissions on the Azure Active Directory to the service principal because it will need to create other service principals for your `AKSCluster`:
+```bash
+# gra
```
Comment (jbw976) [inline docs/cloud-providers/azure/azure-provider.md]: what is this `gra` comment supposed to mean? is it truncated for a more useful comment that starts with `grant...`? :)

File: docs/cloud-providers/azure/azure-provider.md [hunk 2]
```diff
@@ -41,26 +41,17 @@ You might see an error similar to the following, but that is OK, the permissions
...
+az ad app permission admin-consent --id "${AZURE_CLIENT_ID}"
```
Comment (jbw976) [inline docs/cloud-providers/azure/azure-provider.md]: this single step using `az` appears to be much less complicated than the UI sequence, which is awesome.

In your testing, did you do this with a new account that didn't already have these perms enabled? i'm hoping we have confidence that this step is reliable for everyone, as opposed to being vetted only by the account we normally use for testing that already had this set.

Comment (muvaf) [reply docs/cloud-providers/azure/azure-provider.md]: By account, do you mean user account or service principal that is created in the first step?

I think the account I used is an old one with `Global Administrator` role but I created a new service principal in the first step and granted admin-consent to that new service principal with this command. As long as the account that user logged in for `az` command has the `Global Administrator` role, which is mentioned as a note, I think it shouldn't matter. Do you have a different experience?

File: docs/cloud-providers/azure/azure-provider.md (hunk 1 above)
Comment (muvaf) [reply docs/cloud-providers/azure/azure-provider.md]: Thanks for catching! Fixed.

File: docs/cloud-providers/azure/azure-provider.md (hunk 2 above)
Comment (jbw976) [reply docs/cloud-providers/azure/azure-provider.md]: I think you're right about that distinction between user account and service principal, i think your instructions are the streamlined CLI equivalent to what we had previously for GUI steps.  Great!! 🎉

**Pull request merged**